patches/server/0723-Fix-commands-from-signs-not-firing-command-events.patch

```diff
@@ -0,0 +1,103 @@
+  - are logged to the console
+  - fire PlayerCommandPreprocessEvent
+  - work with double-slash commands like `//wand`
+  - sends failure messages to the player who clicked the sign
+
+diff --git a/src/main/java/io/papermc/paper/commands/DelegatingCommandSource.java b/src/main/java/io/papermc/paper/commands/DelegatingCommandSource.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..1f6747d7a4c33f0ee7b0dc2120081bb87a855d35
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/commands/DelegatingCommandSource.java
+@@ -0,0 +1,42 @@
++package io.papermc.paper.commands;
++
++import net.minecraft.commands.CommandSource;
++import net.minecraft.commands.CommandSourceStack;
++import net.minecraft.network.chat.Component;
++import org.bukkit.command.CommandSender;
++
++import java.util.UUID;
++
++public class DelegatingCommandSource implements CommandSource {
++
++    private final CommandSource delegate;
++
++    public DelegatingCommandSource(CommandSource delegate) {
++        this.delegate = delegate;
++    }
++
++    @Override
++    public void sendMessage(Component message, UUID sender) {
++        delegate.sendMessage(message, sender);
++    }
++
++    @Override
++    public boolean acceptsSuccess() {
++        return delegate.acceptsSuccess();
++    }
++
++    @Override
++    public boolean acceptsFailure() {
++        return delegate.acceptsFailure();
++    }
++
++    @Override
++    public boolean shouldInformAdmins() {
++        return delegate.shouldInformAdmins();
++    }
++
++    @Override
++    public CommandSender getBukkitSender(CommandSourceStack wrapper) {
++        return delegate.getBukkitSender(wrapper);
++    }
++}
+diff --git a/src/main/java/net/minecraft/world/level/block/entity/SignBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/SignBlockEntity.java
+index 9b5d11ece006d7aa893360a84ba652c666517ac1..81739da3afd5979ecf745dce39d547c8dcc07250 100644
+--- a/src/main/java/net/minecraft/world/level/block/entity/SignBlockEntity.java
++++ b/src/main/java/net/minecraft/world/level/block/entity/SignBlockEntity.java
+@@ -227,7 +227,15 @@ public class SignBlockEntity extends BlockEntity implements CommandSource { // C
+             ClickEvent chatclickable = chatmodifier.getClickEvent();
+ 
+             if (chatclickable != null && chatclickable.getAction() == ClickEvent.Action.RUN_COMMAND) {
+-                player.getServer().getCommands().performCommand(this.createCommandSourceStack(player), chatclickable.getValue());
++                // Paper start
++                net.minecraft.server.network.ServerGamePacketListenerImpl.LOGGER.info(player.getScoreboardName() + " issued server command: /" + chatclickable.getValue());
++                org.bukkit.event.player.PlayerCommandPreprocessEvent event = new org.bukkit.event.player.PlayerCommandPreprocessEvent(player.getBukkitEntity(), "/" + chatclickable.getValue(), new org.bukkit.craftbukkit.util.LazyPlayerSet(player.getServer()));
```

Comment:
Just logging for future reference: in vanilla, signs on commands *may or may not* have a `/` at the start (ie both `tellraw @p ...` and `/tellraw @p ...` are valid). However, `PlayerCommandPreprocessEvent` always has a leading `/` when fired by `ServerGamePacketListenerImpl#handleCommand`, so this needs to check for the presence of a leading slash and prepend one if missing.

(As for double slash commands from WorldEdit like `//wand`, these never worked prior to #6109, but it makes sense to require two leading slashes to avoid ambiguity.)